apps/entity-service/src/entities/processors/site.processor.ts

```diff
@@ -1,24 +1,167 @@
-import { EntityProcessor, PaginatedResult } from "./entity-processor";
-import { Site } from "@terramatch-microservices/database/entities";
-import { SiteFullDto, SiteLightDto } from "../dto/site.dto";
+import { DocumentBuilder } from "@terramatch-microservices/common/util";
+import { Aggregate, aggregateColumns, EntityProcessor, PaginatedResult } from "./entity-processor";
+import {
+  Demographic,
+  DemographicEntry,
+  Media,
+  ProjectUser,
+  Seeding,
+  Site,
+  SitePolygon,
+  SiteReport,
+  TreeSpecies
+} from "@terramatch-microservices/database/entities";
+import { AdditionalSiteFullProps, SiteFullDto, SiteLightDto, SiteMedia } from "../dto/site.dto";
+import { EntityQueryDto } from "../dto/entity-query.dto";
+import { BadRequestException } from "@nestjs/common";
+import { FrameworkKey } from "@terramatch-microservices/database/constants/framework";
+import { Includeable, Op } from "sequelize";
+import { sumBy } from "lodash";
 
 export class SiteProcessor extends EntityProcessor<Site, SiteLightDto, SiteFullDto> {
   readonly LIGHT_DTO = SiteLightDto;
   readonly FULL_DTO = SiteFullDto;
 
-  async findOne(): Promise<Site> {
-    throw new Error("Method not implemented.");
+  async findOne(uuid: string) {
+    return await Site.findOne({
+      where: { uuid },
+      include: [{ association: "framework" }, { association: "project", attributes: ["uuid"] }]
+    });
   }
 
-  async findMany(): Promise<PaginatedResult<Site>> {
-    throw new Error("Method not implemented.");
+  async findMany(query: EntityQueryDto, userId?: number, permissions?: string[]): Promise<PaginatedResult<Site>> {
+    const projectAssociation: Includeable = {
+      association: "project",
+      attributes: ["name"]
+    };
+    const frameworkAssociation: Includeable = {
+      association: "framework",
+      attributes: ["name"]
+    };
+    if (query.search != null) {
+      // This is they way that sequelize supports for searching in a joined table
+      projectAssociation.where = { name: { [Op.like]: `%${query.search}%` } };
+      // This is to ensure that the project is not required to be joined (simulating an OR)
+      projectAssociation.required = false;
+    }
```

Comment:
Please take a look to this comments, if you feel this is an anti pattern o there is a better way to handle this.
Happy to change it.

Reply:
In this case, this doesn't seem possible.
The second block requires the usage of builder and the first block is required to create the builder.

Reply:
oh, my mistake